src/context.rs

```diff
@@ -600,23 +600,23 @@ impl Context {
         matrix
     }
 
-    pub fn set_font_options(&self, options: FontOptions) {
+    pub fn set_font_options(&self, options: &FontOptions) {
         unsafe {
-            ffi::cairo_set_font_options(self.0, options.get_ptr())
+            ffi::cairo_set_font_options(self.0, options.to_glib_none().0)
         }
     }
 
     pub fn get_font_options(&self) -> FontOptions {
-        let out = FontOptions::new();
+        let mut out = FontOptions::new();
         unsafe {
-            ffi::cairo_get_font_options(self.0, out.get_ptr());
+            ffi::cairo_get_font_options(self.0, out.to_glib_none_mut().0);
```

Comment:
Yes, my point in here is trying to force user to have a mutable access if the underlying data is modifier (unlike get methods/functions).